### ISSUE TITLE
ksonnet 0.10.2 invalid field name "prow_env"

### ISSUE THREAD

**Issue by pdmack** [kubeflow/testing#137]
Possibly related to #124.

/cc @jlewi 

```
INFO|2018-05-22T15:42:44|/src/kubeflow/testing/py/kubeflow/testing/util.py|62| level=info msg="setting environment component" component-name=workflows
INFO|2018-05-22T15:42:45|/src/kubeflow/testing/py/kubeflow/testing/util.py|41| Running: ks param set --env=kubeflow-presubmit-kubeflow-e2e-gke-786-dd63deb-1612-e61e workflows prow_env BUILD_ID=1612,BUILD_NUMBER=1612,JOB_NAME=kubeflow-presubmit,JOB_TYPE=presubmit,PULL_BASE_SHA=692596e912395e7a92f134a80c2b87b2898109f1,PULL_NUMBER=786,PULL_PULL_SHA=dd63deb3102ad9c8a4e5dbfca9555fda4773b636,REPO_NAME=kubeflow,REPO_OWNER=kubeflow 
cwd=/src/kubeflow/kubeflow/testing/workflows
INFO|2018-05-22T15:42:45|/src/kubeflow/testing/py/kubeflow/testing/util.py|56| Subprocess output:

INFO|2018-05-22T15:42:45|/src/kubeflow/testing/py/kubeflow/testing/util.py|62| level=info msg="setting environment component" component-name=workflows
INFO|2018-05-22T15:42:45|/src/kubeflow/testing/py/kubeflow/testing/util.py|62| level=error msg="set params: invalid field name \"prow_env\""
```

**Comment by pdmack** [comment]
kubeflow/kubeflow#786

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/kubeflow_kubeflow/786/kubeflow-presubmit/1612/

**Comment by everpeace** [comment]
Let me apologize for breaking e2e test and bothering you  🙇   I would like to dig the cause.

**Comment by jlewi** [comment]
No apologies needed not your fault we have no way of catching this with presubmits.

**Comment by everpeace** [comment]
hmmm.  this seems to be a bug of ksonnet.  This reproduces in the very simple tutorial example...

```
 $ ks version
ksonnet version: 0.10.2
jsonnet version: v0.10.0
client-go version: 1.8

 $ ks init ks-0.10.2-test
INFO Using context "minikube" from kubeconfig file "/Users/everpeace/.kube/config"
INFO Creating environment "default" with namespace "kube-system", pointing to cluster at address "https://192.168.99.100:8443"
INFO Generating ksonnet-lib data at path '/Users/everpeace/tmp/ks-0.10.2-test/lib/v1.8.0'

$ cd ks-0.10.2-test

# create a smple component from https://ksonnet.io/docs/tutorial
$  ks generate deployed-service guestbook-ui \
>    --image gcr.io/heptio-images/ks-guestbook-demo:0.1 \
>    --type ClusterIP
INFO Writing component at '/Users/everpeace/tmp/ks-0.10.2-test/components/guestbook-ui.jsonnet'

# CASE1:  with `--env` option ==> paremeters with underscore cannot be set....
$ ks param set --env=default guestbook-ui nounderscore "a,b,c"
INFO setting environment component                 component-name=guestbook-ui

$ ks param set --env=default guestbook-ui under_score "a,b,c"
INFO setting environment component                 component-name=guestbook-ui
ERROR set params: invalid field name "under_score"

# CASE2: without `--env` option ==> both parameters can be set.
 $ ks param set guestbook-ui nounderscore "a,b,c"
 $ ks param set guestbook-ui under_score "a,b,c"
```

I will open an issue on ksonnet repo.

**Comment by everpeace** [comment]
I did open a issue on ksonnet repo:  https://github.com/ksonnet/ksonnet/issues/554

**Comment by everpeace** [comment]
Next version of ksonnet will fix this issue. https://github.com/ksonnet/ksonnet/pull/567 fixes above two issues( ksonnet/ksonnet#554 and ksonnet/ksonnet#504)
